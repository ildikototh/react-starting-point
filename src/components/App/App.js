import React from 'react';
import {
  withRouter, Route, Switch,
} from 'react-router-dom';
import NotFound from '../../pages/NotFound/NotFound';
import Main from '../../pages/Main';

const App = () => (
  <Switch>
    <Route exact path="/" component={Main} />
    <Route render={NotFound} />
  </Switch>
);

export default withRouter(App);
