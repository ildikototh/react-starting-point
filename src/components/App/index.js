export { default } from './App';
export { default as App } from './App';
export { default as AppContainer } from './AppContainer';
