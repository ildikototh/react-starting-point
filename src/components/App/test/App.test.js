import React from 'react';
import { shallow } from 'enzyme';
import { App } from '../App';

describe('<App />', () => {
  let location;
  let wrapper;

  beforeEach(() => {
    location = {
      pathname: '',
    };
    wrapper = shallow(<App location={location} />);
  });

  describe('Structure', () => {
    it('should match the snapshot', () => {
      expect(wrapper).toMatchSnapshot();
    });

    it('should include the Routes', () => {
      expect(wrapper).toContainMatchingElement('Switch');
      expect(wrapper).toContainMatchingElements(2, 'Route');
    });
  });
});
