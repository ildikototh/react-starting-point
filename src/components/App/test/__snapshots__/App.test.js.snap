// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<App /> Structure should match the snapshot 1`] = `
<Switch>
  <Route
    component={[Function]}
    exact={true}
    path="/"
  />
  <Route
    render={[Function]}
  />
</Switch>
`;
